src/Hyperledger.Aries/Common/AgentFrameworkException.cs

```diff
@@ -15,7 +18,34 @@ public class AriesFrameworkException : Exception
         /// The error code.
         /// </value>
         public ErrorCode ErrorCode { get; }
-
+   
+        /// <summary>
+        /// Gets the message context record.
+        /// May be <code>null</code>.
+        /// </summary>
+        /// <value>
+        /// The message context record. 
+        /// </value>
+        public RecordBase ContextRecord { get; }
+        
+        /// <summary>
+        /// Gets the connection record.
+        /// May be <code>null</code>.
+        /// </summary>
+        /// <value>
+        /// The connection record. 
+        /// </value>
+        public ConnectionRecord ConnectionRecord { get; }
+        
+        /// <summary>
+        /// Gets the credential record.
+        /// May be <code>null</code>.
+        /// </summary>
+        /// <value>
+        /// The credential record. 
+        /// </value>
+        public CredentialRecord CredentialRecord { get; }
```

Comment:
No need for CredentialRecord. Assign CredentialRecord in the ctor to ContextRecord.